arex-storage-web-api/src/main/java/com/arextest/storage/mock/impl/DefaultMockResultProviderImpl.java

```diff
@@ -68,10 +76,58 @@ public <T extends Mocker> boolean putRecordResult(MockCategoryType category, Str
                 }
             }
         }
+        if (shouldUseIdOfInstanceToMockResult(category) && CollectionUtils.isNotEmpty(unusedRecordInstanceList)) {
+            putRecordInstanceData(category, recordIdBytes, unusedRecordInstanceList);
+        }
         LOGGER.info("put record result to cache size:{} for category:{},record id:{}", size, category, recordId);
         return size > EMPTY_SIZE;
     }
 
+    /**
+     * put RecordInstanceData to cache
+     */
+    private void putRecordInstanceData(MockCategoryType category, byte[] recordIdBytes, List<RecordInstanceData> unusedRecordInstanceList) {
+        byte[] unusedRecordInstanceKey = getRecordInstanceDataKey(category, recordIdBytes);
+        byte[] value = null;
+        try {
+            value = serializer.serialize(objectMapper.writeValueAsString(unusedRecordInstanceList));
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException(e);
+        }
+        redisCacheProvider.put(unusedRecordInstanceKey, cacheExpiredSeconds, value);
+    }
+
+    private byte[] getRecordInstanceDataKey(MockCategoryType category, byte[] recordIdBytes) {
+        return CacheKeyUtils.buildSourceKey(MockResultType.RECORD_INSTANCE_ID, category, recordIdBytes);
+    }
+
+    private byte[] getRecordInstanceData(MockCategoryType category, String recordId) {
+        byte[] recordInstanceDataKey = getRecordInstanceDataKey(category, CacheKeyUtils.toUtf8Bytes(recordId));
+        return redisCacheProvider.get(recordInstanceDataKey);
+    }
+
+    @Override
+    public void updateRecordInstanceStatus(MockCategoryType categoryType, String recordId, int status) {
+        if (!shouldUseIdOfInstanceToMockResult(categoryType)) {
+            return;
+        }
+        byte[] recordInstanceData = getRecordInstanceData(categoryType, recordId);
+        List<RecordInstanceData> allRecordInstanceData = serializer.deserializeToList(recordInstanceData, RecordInstanceData.class);
```

Comment:
> updateRecordInstanceStatus

When multiple users replay the same `record Id` at the same time, concurrent modification issues occur, resulting in mismatched. Recommended to add `replayId`.